renderdoc/api/replay/pipestate.inl

```diff
@@ -1034,31 +1034,38 @@ rdcarray<BoundResourceArray> PipeState::GetReadOnlyResources(ShaderStage stage)
     {
       const D3D12Pipe::Shader &s = GetD3D12Stage(stage);
 
-      size_t size = 0;
-      for(int space = 0; space < s.spaces.count(); space++)
-        size += s.spaces[space].srvs.size();
-
+      size_t size = s.bindpointMapping.readOnlyResources.size();
       ret.reserve(size);
 
-      for(int space = 0; space < s.spaces.count(); space++)
+      for(size_t bp = 0; bp < size; bp++)
       {
-        for(int reg = 0; reg < s.spaces[space].srvs.count(); reg++)
-        {
-          const D3D12Pipe::View &bind = s.spaces[space].srvs[reg];
-          Bindpoint key(s.spaces[space].spaceIndex, reg);
-          BoundResource val;
+        const Bindpoint &bind = s.bindpointMapping.readOnlyResources[bp];
+        ret.push_back(BoundResourceArray());
+        ret.back().bindPoint = bind;
 
-          // consider this register to not exist - it's in a gap defined by sparse root signature
-          // elements
-          if(bind.rootElement == ~0U)
-            continue;
+        uint32_t count = bind.arraySize == ~0U ? 1 : bind.arraySize;
+        ret.back().dynamicallyUsedCount = count;
```

Comment:
D3D12 doesn't support dynamic feedback on resource usage, so `dynamicallyUsedCount` should not be set here. The UI code should be fixed up to handle 0 coming back here if it isn't already.

Reply:
Fixed